src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -135,10 +135,14 @@ void quietlyCloseConnection(final Connection connection, final String closureRea
 
             // continue with the close even if setNetworkTimeout() throws
             try (connection) {
-               if (!connection.isClosed())
+               if (!connection.isClosed()) {
+                  if (!connection.getAutoCommit()) {
```

Comment:
Should there be some check for a dirty state of the connection to avoid rollbacks on all connections closed?

Reply:
When auto commit is true, calling rollback results in an error, so I had to add additional handling.

Reply:
Which is ok IMO, but to avoid rollbacks on every connection closed, can we check for connections to have uncommitted changes before rollback?

Reply:
ok

Reply:
The ProxyConnection knows about the dirtyCommitState and could decide not to call its delegate rollback Method if not dirty.

Reply:
I'm concerned that this method might be called before a ProxyConnection is created through the ProxyFactory.

Reply:
I don't know why the factory would do this and if it is ever a problem to suppress rollbacks if the connection is not marked dirty.

Reply:
I agree with the idea of suppressing rollbacks by checking isCommitStateDirty. However, my concern is that the connection might not be an instance of ProxyConnection. Do you think it would be better to use instanceof to cast and check it?

Reply:
Yes, this could be a way. In case the dynamically created instance fails the instanceOf check it might be better to use isAssignableTo instead.

Reply:
ok, I'll make the revisions as you suggested. However, I have some other commitments, so I might not be able to work on it right away. I plan to complete it within a day or two.

Reply:
Ok.

Reply:
Could you point to the part of the JDBC spec where it says to rollback connections when returning it to the pool? I'm just curious.

Reply:
The quote in https://github.com/brettwooldridge/HikariCP/pull/2238#discussion_r1831436725 is from the JDBC specifications. I based my changes on the suggestion in https://github.com/brettwooldridge/HikariCP/issues/1388#issuecomment-1346172517, so it might be helpful to take a look at it.

Reply:
The second comment  talks about the case of evicting a connection from the pool. In which case a rollback is expected. The method quietlyCloseConnection is called in several more cases, isn't it?

Reply:
After an SQLTimeoutException occurs, quietlyCloseConnection closes the connection, but there’s no rollback, leading to a partial commit. I modified quietlyCloseConnection to address this issue.

If you have concerns, I can skip applying commit 0da3ec366a4d0ad10e4ba1fd174e3dfca1bd7137 and only apply commit 3d17a991a06373a80ee1be9aa787d902a4518c9b instead.

Reply:
I think it is @brettwooldridge who is to decide about the changes ultimately. So for me its good as it is. Thank you for your patience and your work.